File: Src/Backend/SourceCode/TEK.SeatPlan.Entity/Seat.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace TEK.SeatPlan.Entity
{
	public class Seat : TrackedEntity
	{
		[Required]
		public int Number { get; set; }

		[StringLength(255)]
		public string Description { get; set; }

		[Required]
		public bool Active { get; set; }

		[Required]
		public bool Transit { get; set; }

		[Required]
		public int Row { get; set; }

		[Required]
		public int Col { get; set; }

		[Required]
		public int LocationId { get; set; }

		public virtual Location Location { get; set; }

		public int? EmployeeId { get; set; }

		public virtual Employee Employee { get; set; }

		public virtual IEnumerable<SeatChangeLog> SourceSeatChangeLog { get; set; }

		public virtual IEnumerable<SeatChangeLog> TargetSeatChangeLog { get; set; }
	}
}

File: Src/Frontend/src/app/core/models/seat.ts
import {Employee} from './employee';

export class Seat {
  number: number;
  transit: boolean;
  description: string;
  row: string;
  col: string;
  employee: Employee;
  locationId: number;

  constructor(number: number, transit: boolean, description: string, employee: Employee, row: string, col: string, locationId: number) {
    this.number = number;
    this.transit = transit;
    this.description = description;
    this.employee = employee;
    this.row = row;
    this.col = col;
    this.locationId = locationId;
  }
}

export class TransitSeat extends Seat {
  constructor(seat: Seat) {
    super(seat.number, seat.transit, seat.description, seat.employee, 'auto', 'auto', 1);
  }
}

export class OfficeSeat extends Seat {
  constructor(seat: Seat) {
    super(seat.number, seat.transit, seat.description, seat.employee, seat.row + 'px', seat.col + 'px', seat.locationId);
  }
}
